x-pack/plugins/security_solution/public/common/components/threat_match/entry_item.tsx

```diff
@@ -55,11 +60,27 @@ export const EntryItem: React.FC<EntryItemProps> = ({
     [onChange, entry]
   );
 
+  const dataViewInstance = useMemo(() => {
+    if (indexPattern != null) {
+      const dv = new DataView({ spec: indexPattern, fieldFormats });
+      return { ...dv, fields: Object.values(indexPattern.fields ?? {}) } as DataView;
```

Comment:
is this possible to avoid type castings?

Reply:
This is a great question - I kept these type castings because until we also update all the exceptions code to accept the `DataViewSpec` type, we can't update the child component `FieldComponent`, so once that work is completed we can remove the type castings all together which will be great. I'll add a todo to the code.